### PR TITLE
Ivan's PR with specs files only for HW_05 as HW_06

### PR DESCRIPTION
**Pull request by BeOff**

### CODE REVIEW

File: HW_06/Ivan_Turliuk/le_git_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+context 'when accounting homeworks' do
+  it { expect {subject.keep_track}.to_not raise_error }
+end
+context 'when prints homeworks list' do
+  it { expect {subject.homework_list}.to output("Any homeworks.\n").to_stdout }
+end
+describe '#post_request' do
+  before { allow($stdout).to receive(:write) }
+  it 'makes a post request' do
+    expect(subject.post_request.is_a?(Net::HTTPOK)).to eq true
```
Comment (labs-ruby) [inline HW_06/Ivan_Turliuk/le_git_spec.rb]: Use `webmock` to check external APIs calls

File: HW_06/Ivan_Turliuk/le_git_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  end
+end
+describe '#notifications' do
+  it 'adds notification if subscribed' do
+    expect(subject.notifications).to eq false
+  end
+end
+context 'when gets check request' do
```
Comment (labs-ruby) [inline HW_06/Ivan_Turliuk/le_git_spec.rb]: Basically, this is not a context but a describe block
`describe #check_request`

File: HW_06/Ivan_Turliuk/le_git_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  it { expect {subject.check_request}.to output("Nothing to check.\n").to_stdout }
+end
+context 'when subscribing to student' do
+  it { expect {subject.subscribe_to_student}.to output("Now subscribed to default.\n").to_stdout }
```
Comment (labs-ruby) [inline HW_06/Ivan_Turliuk/le_git_spec.rb]: The same, incorrect usage of contexts

File: HW_06/Ivan_Turliuk/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+RSpec.describe Student do
+  subject { Student.new(pid: 'Test Student') }
+  describe '#do_homework' do
+    before do
+      allow($stdout).to receive(:write)
+      allow(subject).to receive(:gets).and_return("\n")
+    end
+    it 'makes student do homework' do
+      expect(subject.do_homework.class).to eq Homework
```
Comment (labs-ruby) [inline HW_06/Ivan_Turliuk/student_spec.rb]: Why don't you check that homework created with correct params? I see the following params in your code:
```
what: what, who: @pid, note: note
```

File: HW_06/Ivan_Turliuk/le_git_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+end
+describe '#post_request' do
+  before { allow($stdout).to receive(:write) }
+  it 'makes a post request' do
+    expect(subject.post_request.is_a?(Net::HTTPOK)).to eq true
+  end
+end
+describe '#notifications' do
+  it 'adds notification if subscribed' do
+    expect(subject.notifications).to eq false
```
Comment (labs-ruby) [inline HW_06/Ivan_Turliuk/le_git_spec.rb]: Don't understand the spec.
The title says that it adds a notification, however you comparing notifications array with `false`. Why?

Comment (BeOff) [reply HW_06/Ivan_Turliuk/le_git_spec.rb]: Corrected title, added requested check.

File: HW_06/Ivan_Turliuk/student_spec.rb (hunk 4 above)
Comment (BeOff) [reply HW_06/Ivan_Turliuk/student_spec.rb]: Added checks for parameters.

File: HW_06/Ivan_Turliuk/le_git_spec.rb (hunk 3 above)
Comment (BeOff) [reply HW_06/Ivan_Turliuk/le_git_spec.rb]: Rewrote with 'describe'.

File: HW_06/Ivan_Turliuk/le_git_spec.rb (hunk 2 above)
Comment (BeOff) [reply HW_06/Ivan_Turliuk/le_git_spec.rb]: Rewrote with 'describe'.

File: HW_06/Ivan_Turliuk/le_git_spec.rb (hunk 1 above)
Comment (BeOff) [reply HW_06/Ivan_Turliuk/le_git_spec.rb]: Rewrote using `webmock`.

**Pull request closed without merging**